### PR TITLE
390 timer database names

### LINKED ISSUE #390: Exam: Timer Database Names
### Description:

Change exam time fields in Firebase to better reflect their purpose.

### Acceptance Criteria:
- [ ] startTime --> openTime.
- [ ] endTime --> closeTime.
- [ ] All functionality is still working.

### Definition of Done:
- [ ] AC met
- [ ] PR linked
- [ ] Branch merged

### PR DESCRIPTION
**Pull request by nzkevc**
**Issue**

The names for exam fields "startTime" and "endTime" don't accurately reflect their purpose, which is to set when the exam opens and closes, rather than when the exam has been sat and submitted.

Closes #390 

**Solution**

Rename every reference of "startTime" and "endTime" in the repo to "openTime" and "closeTime". So:

- startTime --> openTime
- endTime --> closeTime

**Risks**

Old quizzes with old names might have to be deleted?

**Reviewers**

@bcho892 
@Oculux314

### CODE REVIEW

File: service/src/schemas/input.graphql
Comment (bcho892) [inline service/src/schemas/input.graphql:36]: can we make all of these non-compulsory (remove all of the `!`)

Comment (bcho892) [reply service/src/schemas/input.graphql:36]: only in this file btw

File: service/src/controllers/quiz.ts
Comment (bcho892) [inline service/src/controllers/quiz.ts:23]: can we make all of these non- compulsory (add a `?` after all of them for `addQuiz`) to avoid merge conflicts

**Review by bcho892 — CHANGES_REQUESTED** [review]
looks mostly fine to me, just a couple of changes to prevent merge conflicts later on.

File: service/src/schemas/input.graphql
Comment (nzkevc) [reply service/src/schemas/input.graphql]: Okay I copied exactly what you did from the other branch so should be good

**Review by Oculux314 — CHANGES_REQUESTED** [review]
Do these remaining instances also need to be renamed?

image
image

**Comment by nzkevc** [comment]
> Do these remaining instances also need to be renamed?
> 
> image image

Oh dang yeah for the yarn file, no. But for List.vue, yeah probably. Thanks for catching that! 

On another note, I wrote one-line to fix the addUserQuiz mutation (it didn't work while I was testing because the submitted field wasn't being set to anything, so firebase was complaining about undefined fields) - so the fix was to just set submitted to false when adding (enrolling) someone to a quiz.

**Review by bcho892 — APPROVED** [review]

**Review by bcho892 — APPROVED** [review]

**Review by Oculux314 — APPROVED** [review]
Looks great!

Tangentially-related notes (not required for approval tho):
- Submitted flag should be timestamp, not boolean
- Timer starts shows 00:00:00 briefly when opening exam

**Pull request merged**
